File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
        int[] ans = new int[2];
        for(int i = 0; i < nums.length; i++) {
            if(hm.containsKey(target - nums[i])) {
                int ind1 = i;
                int ind2 = hm.get(target - nums[i]);
                
                ans[0] = ind1;
                ans[1] = ind2;
                
                return ans; 
            } else {
                hm.put(nums[i], i);
            }
        }
        return ans; 
    }
}

File: 1-two-sum/1-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map: HashMap<Int, Int> = HashMap()
        for ((index, num) in nums.withIndex()) {
            map.put(nums[index], index)
        }
        for ((index, num) in nums.withIndex()) {
            val complement = target - nums[index]
            if (map.containsKey(complement) && 
                    map[complement] != index)             {
                return listOf(
                    index, 
                    map[complement] ?: index + 1
                ).toIntArray()
            }
        }
        return nums
    }
}